core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2305MultipleProxiesTest.java

```diff
@@ -71,20 +78,16 @@ public void testit()
         String keyPwd = "key-passwd";
 
         Server server = new Server( 0 );
-        server.addConnector( newHttpsConnector( storePath, storePwd, keyPwd ) );
+        addHttpsConnector( server, storePath, storePwd, keyPwd );
         server.setHandler( new RepoHandler() );
         server.start();
-        while ( !server.isRunning() || !server.isStarted() )
+        if ( server.isFailed() )
         {
-            if ( server.isFailed() )
-            {
-                fail( "Couldn't bind the server socket to a free port!" );
-            }
-            Thread.sleep( 100L );
+            fail( "Couldn't bind the server socket to a free port!" );
         }
-        int httpPort = server.getConnectors()[0].getLocalPort();
+        int httpPort = ( (NetworkConnector) server.getConnectors()[0] ).getLocalPort();
         System.out.println( "Bound server socket to HTTP port " + httpPort );
-        int httpsPort = server.getConnectors()[1].getLocalPort();
+        int httpsPort = ( (NetworkConnector) server.getConnectors()[1] ).getLocalPort();
```

Comment:
The `HttpConfiguration` belonging to Connector[0] needs to have its `HttpConfiguration.setSecurePort(int)` set for the port that this HTTPS connector.

Reply:
The thing is that we do not set the port, via `HttpConfiguration.setSecurePort(int)`, because we do not know it. We can only read both ports as local ports.

Reply:
Then your Requests never indicate that they are secure.
The Requests will not contain the security details.
The Requests will not have the javax attributes for the SSL / Certificate state of the connection.
Nor will you have ability to use the "confidential" constraints.
And usages that redirect from HTTP to HTTPS will also fail. (such as SecureRedirectHandler)

Reply:
@joakime
I am convinced your advice is right. Can you improve these tests and publish a new PR?